src/Nncase.Passes/Rules/Neutral/AddPreProcess.cs

```diff
@@ -175,13 +175,28 @@ protected override Task<IRModule> RunCoreAsync(IRModule module, RunPassContext o
             // Normalization
             if (mean.Length != 0)
             {
-                newInput = mean.Length switch
+                switch (mean.Length)
                 {
-                    3 when inputShape.Length == 4 => (newInput - Tensor.From(mean, new[] { 1, mean.Length, 1, 1 })) /
-                                                     Tensor.From(std, new[] { 1, std.Length, 1, 1 }),
-                    _ => (newInput - Tensor.From(new float[] { mean[0] }, new[] { 1 })) /
-                         Tensor.From(new float[] { std[0] }, new[] { 1 }),
-                };
+                    case 3 when inputShape.Length == 4:
+                        var meanCall = (Expr)Tensor.From(mean, new[] { 1, mean.Length, 1, 1 });
+                        meanCall.Metadata.OutputNames = new[] { "Mean" };
+                        var stdCall = (Expr)Tensor.From(std, new[] { 1, std.Length, 1, 1 });
+                        stdCall.Metadata.OutputNames = new[] { "Std" };
+                        var subMean = (newInput - meanCall).With(metadata: new IRMetadata() { OutputNames = new[] { input.Metadata.OutputNames?[0] + "_SubMean" } });
+                        var divStd = (subMean / stdCall).With(metadata: new IRMetadata() { OutputNames = new[] { input.Metadata.OutputNames?[0] + "_DivStd" } });
+                        newInput = divStd;
+                        break;
+
+                    default:
+                        meanCall = (Expr)Tensor.From(new float[] { mean[0] }, new[] { 1 });
+                        meanCall.Metadata.OutputNames = new[] { "Mean" };
```

Comment:
It's better that assign op name once after switch.

Reply:
I have changed the implementation method according to your opinion, and it feels very good. Please help me review it again